audiobook-reading-order-ids/README.md

```diff
@@ -0,0 +1,257 @@
+every `templated` `href` value to be the same, and for each reading order item to contain a
+`properties` object containing values to be substituted into the `href` in order to produce a
+unique URI. This is a level of complexity that we currently do not want to mandate that players
+handle correctly, due to the apparent almost nonexistent use of `templated` links.
+
+### Findaway
+
+Manifests in Findaway audiobooks loosely adhere to the WebPub manifest specification except for
+the fact that they typically do not provide any kind of `href` value in `readingOrder` arrays
+at all. Instead, each `readingOrder` item will contain the proprietary, integer-typed extension
+properties `findaway:part` and `findaway:sequence`. These integer-typed values are accepted by
+the proprietary AudioEngine player to select chapters. In any given manifest, the combination
+of `findaway:part` and `findaway:sequence` values can be trusted to be unique.
+
+The identifier that applications should generate for a reading order item with
+`findaway:part = p` and `findaway:sequence = s` is:
+
+```
+urn:org.thepalaceproject:findaway:p:s
+```
+
+Newer manifests _may_ include `href` values. If the manifest does contain a non-`templated` 
+`href` value, that value should be used in favour of this generation scheme (as, in that case, 
+the manifest is essentially a valid [WebPub manifest](#webpub-manifests)).
+
+### Overdrive
+
+Manifests from Overdrive are in a completely proprietary JSON format. The manifests contain a
+`contentlinks` array that looks similar to:
+
+```
+"contentlinks" : [
+ {
+    "href" : "https://example.com/data?body=tokentokentokentokentokenQwMS5tcDMifX0%3D&s=ess",
+    "type" : "text/html",
+    "physicalFileLengthInBytes" : 35399262
+ },
+ {
+    "physicalFileLengthInBytes" : 26515957,
+    "type" : "text/html",
+    "href" : "https://example.com/data?body=tokentokentokentokentokenQwMi5tcDMifX0%3D&s=ess"
+ },
+```
+
+Each `href` property inside each `contentlinks` element has a value that is a URI that has a 
+short expiration date. Therefore, the values of these `href` properties are not _stable_ and
+therefore cannot be used to derive stable identifiers. As none of the other properties within
+the `contentlinks` elements will uniquely identify that element, the only option applications
+have is to refer to elements by integer index.
+
+We therefore number each element of the `contentlinks` array, starting at `0`, and
+the identifier that applications should generate for a reading order item with integer index `i`
+is equal to that of the [Fallback](#fallback) rule for a reading order item `i`.
+
+### Fallback
+
+If none of the above rules apply for the current manifest, then the application must resort to
+generating identifiers based on an integer index into an array of reading order items. For
+a reading order item index `i`, starting at `0`, the identifier generated must be:
+
+```
+urn:org.thepalaceproject:reading_order_item:i
```

Comment:
Super minor, but I think camel case is preferred in URI, so maybe `readingOrder` would be more appropriate? NBD either way though 🤷🏻

Reply:
:shrug: 

:laughing:

Reply:
Might as well change this for consistency with the way WebPub manifests spell it.